### DISCUSSION TITLE
Are these images hosted anywhere?

### DISCUSSION THREAD

**Discussion by irl-segfault** [actions/virtual-environments#7363] [General questions] [answered]
Hi all,

I'd like to `docker run` the default Ubuntu image used in Github Actions, but I don't see a mention of where that's hosted, or instructions for running. I do see instructions for building, but would prefer to just `docker run $X`.  Thanks in advance.

**Comment by erik-bershel** [comment, marked as answer]
Hey @irl-segfault!
We don't have our images as docker images.

**Comment by MichaelBelousov** [comment]
related question, are there any scripts to provision a VM from the image descriptions?  
I have an obscure failure occurring only in github windows server runners, and it would be great to be able to reproduce that on my own hardware to be able to inspect it better.

**Reply by erik-bershel** [reply]
We provide a script that allows you to prepare a virtual machine using Packer. You can modify it for your specific use case or run necessary scripts manually. Readme: https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md

**Reply by romanr** [reply]
What about macOs? That document is only about  Linux or Windows.

**Reply by pkit** [reply]
@romanr MacOS is done on Anka Cloud. Not really useful.

**Comment by dlmiles** [comment]
Can we access the container registry from inside a github-action runner, download a copy of itself and upload it to somewhere else on the internet ?

**Reply by pkit** [reply]
No, as it's a VM image. And it's 51GB compressed.

**Reply by jyling** [reply]
Where can i download the VM image ?

**Reply by pkit** [reply]
Nowhere, you can only build it yourself. Like I did.

**Reply by jyling** [reply]
Ah, i see, thanks

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
